src/App.tsx

```diff
@@ -1,223 +1,153 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  FormEvent,
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import { createTodo, deleteTodo, getTodos } from './api/todos';
 import { AuthContext } from './components/Auth/AuthContext';
+import { Errors } from './components/Errors';
+import { Footer } from './components/Footer';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { FilterType } from './types/FilterType';
+import { Todo } from './types/Todo';
+
+export function filterTodos(
+  todos: Todo[],
+  filterType: string,
+) {
+  const filteredTodos = todos.filter((todo) => {
+    switch (filterType) {
+      case 'completed':
+        return todo.completed;
+
+      case 'active':
+        return !todo.completed;
+
+      default:
+        return todo;
+    }
+  });
+
+  return filteredTodos;
+}
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filteredBy, setFilteredBy] = useState<FilterType>(FilterType.All);
+  const [title, setTitle] = useState('');
+  const [errorText, setErrorText] = useState('');
+  const [selectedId, setSelectedId] = useState<number[]>([]);
+  const [isAdding, setIsAdding] = useState(false);
+
+  const visibleTodos = filterTodos(todos, filteredBy);
 
   useEffect(() => {
+    const todosFromServer = async () => {
+      if (user) {
+        const getedTodos = getTodos(user.id);
+
+        setTodos(await getedTodos);
+      }
+    };
+
+    todosFromServer();
+
     // focus the element with `ref={newTodoField}`
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
   }, []);
 
+  const addTodo = useCallback(async (event: FormEvent) => {
+    event.preventDefault();
+    if (!title.trim || !user) {
+      setErrorText("You can't do something if you don't know what exactly.");
+
+      return;
+    }
+
+    setIsAdding(true);
+
+    try {
+      const postTodo = await createTodo(title, user.id);
+
+      setTodos((prevTodos) => [...prevTodos, postTodo]);
+    } catch {
+      setErrorText('123');
```

Comment:
interesting code of error 🤔 
Do users have to install or uninstall something from your page so they get such an error code?

Reply:
or maybe they are denied because of pure permissions?
Make your error test more informative! use code only for consoles aca for other devs. and for user you have to show the reason in text message)